src/types.rs

```diff
@@ -269,6 +271,12 @@ pub struct OneExit {
     pub provider_name: String,
     /// The homepage of the provider of this server.
     pub provider_homepage_url: String,
+    /// ID of the datacenter server is located at.
+    #[serde(default = "random_string")]
+    pub datacenter_id: String,
```

Comment:
Nit: can we just make this a u64 or similar? We don't need much collision detection even if we generate with a hash.

Reply:
Unless we are happy with u32 (fine by me) that could be problematic for JS.